src/ui/app.ts

```diff
@@ -91,20 +95,27 @@ class ArchiveWebApp extends ReplayWebApp {
       this.btrixOpts = null;
     }
 
-    if (!self.localStorage.getItem("archiveCookies")) {
-      self.localStorage.setItem("archiveCookies", "1");
-    }
-
-    getLocalOption("autorunBehaviors").then(
-      (res) => (this.autorun = res === "1"),
-    );
-
     if (window.archivewebpage) {
       // @ts-expect-error - TS7006 - Parameter 'progress' implicitly has an 'any' type.
       window.archivewebpage.setDownloadCallback((progress) =>
         this.onDownloadProgress(progress),
       );
     }
+
+    void this.initOpts();
+  }
+
+  async initOpts() {
+    this.autorun = (await getLocalOption("autorunBehaviors") === "1");
+
+    const archiveCookies = await getLocalOption("archiveCookies");
+    // default to true if unset to match existing behavior
+    if (archiveCookies === null) {
+      await setLocalOption("archiveCookies", "1");
+      this.archiveCookies = true;
+    }
```

Comment:
Might need to add another `if` statement here to set this.archiveCookies to true if `archiveCookies === "1"`?